src/BuildCheck.UnitTests/EndToEndTests.cs

```diff
@@ -105,6 +105,36 @@ public void SampleAnalyzerIntegrationTest_ReplayBinaryLogOfAnalyzedBuild(bool bu
         }
     }
 
+    [Theory]
+    [InlineData("warning", "warning BC0101")]
+    [InlineData("error", "error BC0101")]
+    [InlineData("suggestion", "BC0101")]
+    [InlineData("default", "warning BC0101")]
+    public void EditorConfig_SeverityAppliedCorrectly(string BC0101Severity, string expectedOutputValues)
+    {
+        PrepareSampleProjectsAndConfig(true, out TransientTestFile projectFile, BC0101Severity);
+
+        string output = RunnerUtilities.ExecBootstrapedMSBuild(
+            $"{Path.GetFileName(projectFile.Path)} /m:1 -nr:False -restore -analyze",
+            out bool success, false, _env.Output, timeoutMilliseconds: 120_000);
```

Comment:
Not necessarily to be aprt of this PR - we should remove the timeout explicit parameter (here and in the other tests) and rely on sane default of 30 seconds.
2 minutes seems excessive here. I know it is from the other tests - but that was added as a temporary desperate measure for the randomly staling integration test - which is now resolved.

Reply:
If we'd add `string[] unexpectedOutputValues` argument to the parametrized test - than we'd be able to more reliably test the `suggestion` case, as well as include the `none` case here and avoid duplication.

Reply:
I was thinking to do some refactoring there in this PR, but decided to do that after this is merged, overall agree. 

Thank you for the suggestion `unexpectedOutputValues` working on that